python/lsst/analysis/tools/tasks/reconstructor.py

```diff
@@ -95,3 +97,46 @@ def reconstructAnalysisTools(
         inputs = callback(inputs, dataId)
 
     return (config, inputs)
+
+
+def getPlotDatasetTypeNames(
+    butler: Butler,
+    collections: str | Iterable[str],
+    label: str | None = None,
+) -> Iterable[str]:
+    """Query the butler to determine dataset type names for analysis tools
+    plots inside a collection or list of collections.
+
+    Parameters
+    ----------
+    butler : `~lsst.daf.butler.Butler`
+        The butler where the data is stored.
+    collections : `str` or `list` [`str`]
+        Collection(s) within the butler associated with the desired data.
+    label : `str`, optional
+        The label from the `~lsst.pipe.base.Pipeline` associated with the task
+        who's plots are to be queried. If no label is given, the entire
+        collection will be queried.
+
+    Yields
+    ------
+    plotName : `str`
+        Dataset type name of the plot.
+    """
+    if isinstance(collections, str):
+        collections = [collections]
```

Comment:
You don't need this here: `butler.get` and `queryDatasets` both handle it for you.